aesara/link/numba/dispatch/scalar.py

```diff
@@ -29,6 +30,10 @@
 )
 
 
+# numba_njit = partial(numba.njit, cache=config.numba__cache)
+numba_njit = numba.njit
```

Comment:
Trying to cache the `Scalar` `Op`s raises an: Can't pickle `PyCapsule` objects error from within Numba.

Reply:
Why is that?

Reply:
Seems like some issue with the way we're compiling string encoded functions:

```python
import numba
from aesara.link.utils import compile_function_src

a = 10.0
b = 20.0

fn_src = """
def some_fn(x, y):
    return x + y
"""

some_fn = compile_function_src(fn_src, "some_fn")

signature = numba.types.float64(numba.types.float64, numba.types.float64)

fn = numba.njit(signature, cache=True)(
    some_fn
)

# Above line fails with
# Traceback (most recent call last):
#   File "/home/kc611/Desktop/Workspaces/Aesara WorkSpace/issues/addnumbacache/pycapsule_issue.py", line 20, in <module>
#     fn = numba.njit(signature, cache=True)(
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/decorators.py", line 226, in wrapper
#     disp.compile(sig)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/dispatcher.py", line 986, in compile
#     self._cache.save_overload(sig, cres)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/caching.py", line 661, in save_overload
#     self._save_overload(sig, data)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/caching.py", line 671, in _save_overload
#     self._cache_file.save(key, data)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/caching.py", line 489, in save
#     self._save_data(data_name, data)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/caching.py", line 547, in _save_data
#     data = self._dump(data)
#   File "/home/kc611/anaconda3/envs/aesara/lib/python3.8/site-packages/numba/core/caching.py", line 560, in _dump
#     return pickle.dumps(obj, protocol=-1)
# TypeError: cannot pickle 'PyCapsule' object
```

Reply:
The issue appears to be caused by the `__global__`s (e.g. perhaps inherited from `aesara.link.utils`).

If you change `some_fn = compile_function_src(fn_src, "some_fn")` to `some_fn = compile_function_src(fn_src, "some_fn", globals=globals())` it will work.

Reply:
I just created an issue for this https://github.com/numba/numba/issues/7641.